presto-main/src/main/java/io/prestosql/sql/planner/PlanFragmenter.java

```diff
@@ -604,6 +608,33 @@ public GroupedExecutionProperties visitAggregation(AggregationNode node, Void co
             return GroupedExecutionProperties.notCapable();
         }
 
+        @Override
+        public GroupedExecutionProperties visitWindow(WindowNode node, Void context)
+        {
+            return visitWindowFunctionNode(node);
+        }
+
+        @Override
+        public GroupedExecutionProperties visitRowNumber(RowNumberNode node, Void context)
+        {
+            return visitWindowFunctionNode(node);
+        }
+
+        @Override
+        public GroupedExecutionProperties visitTopNRowNumber(TopNRowNumberNode node, Void context)
+        {
+            return visitWindowFunctionNode(node);
+        }
+
+        private GroupedExecutionProperties visitWindowFunctionNode(PlanNode node)
```

Comment:
I'd call this method something else to make it clear it's *not* part of the visitor interface. Maybe `processWindowFunction`.